build-logic/gradle/src/main/kotlin/com/avito/android/GenerateCommonProperties.kt

```diff
@@ -0,0 +1,53 @@
+package com.avito.android
+
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputDirectories
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+abstract class GenerateCommonProperties : DefaultTask() {
+
+    @get:InputFile
+    abstract val commonPropertiesFile: RegularFileProperty
+
+    @get:OutputDirectories
+    abstract val projectDirs: SetProperty<Directory>
+
+    @TaskAction
+    fun doWork() {
+        val commonProperties = PropertiesConfiguration(commonPropertiesFile.get().asFile)
+
+        projectDirs.get().forEach { projectDir ->
+
+            val gradlePropertiesFile = File(projectDir.asFile, "gradle.properties")
+
+            val properties = if (gradlePropertiesFile.exists()) {
+                PropertiesConfiguration(gradlePropertiesFile)
+            } else {
+                PropertiesConfiguration(gradlePropertiesFile)
+            }
```

Comment:
Why are files in `if` and `else` the same?

```kotlin
PropertiesConfiguration(gradlePropertiesFile)
```

Reply:
Thanks, first implementation was using `java.util.Properties` API , but `commons-configuration` doesn't care